pkg/storegateway/stats.go

```diff
@@ -110,6 +111,11 @@ func newBlockQueriedMeta(meta *block.Meta) blockQueriedMeta {
 	if meta.Compaction.Level == 0 {
 		m.level = "unknown/old_block"
 	}
+
+	// The compactor doesn't preserve the "FromOutOfOrder" hints. Instead, we should check for the external label, that Mimir sets.
+	if !m.outOfOrder && meta.Thanos.Labels != nil {
+		m.outOfOrder = meta.Thanos.Labels[mimir_tsdb.OutOfOrderExternalLabel] == mimir_tsdb.OutOfOrderExternalLabelValue
+	}
```

Comment:
should this be something like a method on the block.meta? i can see it being confusing to other people too